packages/devextreme-react/src/core/__tests__/component.test.tsx

```diff
@@ -1,137 +1,152 @@
 import * as events from 'devextreme/events';
-
-import { mount, React, shallow } from './setup';
+import { render, cleanup } from '@testing-library/react';
+import * as React from 'react';
 import {
-  fireOptionChange, TestComponent, Widget, WidgetClass,
+  fireOptionChange,
+  TestComponent,
+  Widget,
+  WidgetClass,
 } from './test-component';
 
 describe('rendering', () => {
+
+  afterEach(() => {
+    WidgetClass.mockClear();
+    cleanup();
+  })
   it('renders correctly', () => {
-    const component = shallow(
+    const { container } = render(
       <TestComponent />,
     );
-    expect(component.type()).toBe('div');
-  });
 
-  it('create widget on componentDidMount', () => {
-    shallow(
-      <TestComponent />,
-    );
+    const element: HTMLElement|null = container.firstChild as HTMLElement;
 
-    expect(WidgetClass.mock.instances.length).toBe(1);
+    expect(element.tagName.toLowerCase()).toBe('div');
   });
 
-  it('pass templatesRenderAsynchronously to widgets', () => {
-    shallow(
-      <TestComponent />,
-    );
+it('create widget on componentDidMount', () => {
+  render(<TestComponent />)
 
-    expect(WidgetClass.mock.calls[0][1]).toEqual({ templatesRenderAsynchronously: true });
-  });
 
-  it('creates nested component', () => {
-    mount(
-      <TestComponent>
-        <TestComponent />
-      </TestComponent>,
-    );
+  expect(WidgetClass.mock.instances.length).toBe(1);
+});
 
-    expect(WidgetClass.mock.instances.length).toBe(2);
-    expect(WidgetClass.mock.instances[1]).toEqual({});
-  });
+it('pass templatesRenderAsynchronously to widgets', () => {
+  render(
+    <TestComponent />,
+  );
 
-  it('do not pass children to options', () => {
-    mount(
-      <TestComponent>
-        <TestComponent />
-      </TestComponent>,
-    );
+  expect(WidgetClass.mock.calls[0][1]).toEqual({ templatesRenderAsynchronously: true });
+});
 
-    expect(WidgetClass.mock.calls[1][1].children).toBeUndefined();
-  });
+it('creates nested component', () => {
+  render(
+    <TestComponent>
+      <TestComponent />
+    </TestComponent>
+  )
+
+  expect(WidgetClass.mock.instances.length).toBe(2);
+  expect(WidgetClass.mock.instances[1]).toEqual({});
+});
+
+it('do not pass children to options', () => {
+  render(
+    <TestComponent>
+      <TestComponent />
+    </TestComponent>,
+  );
+
+  expect(WidgetClass.mock.calls[1][1].children).toBeUndefined();
+});
 });
 
 describe('element attrs management', () => {
   it('passes id, className and style to element', () => {
-    const component = mount(
-      <TestComponent id="id1" className="class1" style={{ background: 'red' }} />,
-    );
+    const { container } = render(
+      <TestComponent id="id1" className="class1" style={{ background: 'red' }} />, {
+    });
 
-    const node = component.getDOMNode();
+    const element: HTMLElement|null = container.firstChild as HTMLElement;
```

Comment:
It seems that type null is exccessive.
```suggestion
    const element: HTMLElement = container.firstChild as HTMLElement;
```